core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -244,5 +244,13 @@ private void resolvePipelineAggregatorOrder(Map<String, AggregatorFactory> aggFa
                 orderedPipelineAggregators.add(factory);
             }
         }
+
+        AggregatorFactory[] getAggregatorFactories() {
+            return this.factories.toArray(new AggregatorFactory[this.factories.size()]);
+        }
+
+        List<PipelineAggregatorFactory> getPipelineAggregatorFactories() {
```

Comment:
I agree and would prefer to use a list everywhere for aggs but I would rather not tackle that in this PR. I think the idea is that its a list here in the Builder as it needs to be dynamic so you can call addAggregator() but in the build() method it becomes an array because it doesn't change after that.